include/crow/http_response.h

```diff
@@ -209,21 +208,54 @@ namespace crow
                 {
                     std::vector<asio::const_buffer> buffers;
                     buffers.push_back(boost::asio::buffer(buf));
-                    boost::asio::write(adaptor.socket(), buffers, [this](std::error_code ec, std::size_t)
+                    write_buffer_list(buffers, adaptor);
+                }
+            }
+
+            //THIS METHOD DOES MODIFY THE BODY, AS IN IT EMPTIES IT
+            template<typename Adaptor>
+            void write_streamed_string(std::string& is, Adaptor& adaptor)
+            {
+                std::string buf;
+                buf.reserve(16384);
+
+                //TODO this can be optimized
+                while (is.length() > 0)
+                {
+                    std::vector<asio::const_buffer> buffers;
+                    if (is.length() > 16384)
+                    {
+                        buf = is.substr(0, 16384);
+                        is = is.substr(16384);
+                    }
+                    else
                     {
-                        if (!ec)
-                        {
-                            return false;
-                        }
-                        else
-                        {
-                            CROW_LOG_ERROR << ec << " - happened while sending body";
-                            this->end();
-                            return true;
-                        }
-                    });
+                        buf = std::string();
```

Comment:
Why set it to new object and then assign `is`?

Reply:
it seems one of my debugging lines made it here, my bad